packages/block-editor/src/components/page-template-picker/picker.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+import { parse } from '@wordpress/blocks';
+import { withDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import __experimentalBlockListFooter from '../block-list-footer';
+import Button from './button';
+import Container from './container';
+import defaultTemplates from './default-templates';
+
+const __experimentalPageTemplatePicker = ( { templates = defaultTemplates, resetContent } ) => {
+	return (
+		<__experimentalBlockListFooter>
+			<Container style={ { flexDirection: 'row' } }>
```

Comment:
Another possibility I considered is to remove the `Container` here, and wrap the slot in `block-list.native.js` instead, so we can avoid the platform divergence here.

<details>
<summary>For example, this seems to work as well</summary>

```diff
diff --git a/packages/block-editor/src/components/block-list/index.native.js b/packages/block-editor/src/components/block-list/index.native.js
index 9230407..a453721 100644
--- a/packages/block-editor/src/components/block-list/index.native.js
+++ b/packages/block-editor/src/components/block-list/index.native.js
@@ -164,7 +164,9 @@ export class BlockList extends Component {
 				} } >
 					<View style={ styles.blockListFooter } />
 				</TouchableWithoutFeedback>
-				<__experimentalBlockListFooter.Slot />
+				<View style={ { flexDirection: 'row' } }>
+					<__experimentalBlockListFooter.Slot />
+				</View>
 			</>
 		);
 	}
diff --git a/packages/block-editor/src/components/page-template-picker/container.js b/packages/block-editor/src/components/page-template-picker/container.js
deleted file mode 100644
index de666e9..0000000
--- a/packages/block-editor/src/components/page-template-picker/container.js
+++ /dev/null
@@ -1,9 +0,0 @@
-const Container = ( { children } ) => {
-	return (
-		<>
-			{ children }
-		</>
-	);
-};
-
-export default Container;
diff --git a/packages/block-editor/src/components/page-template-picker/container.native.js b/packages/block-editor/src/components/page-template-picker/container.native.js
deleted file mode 100644
index d987d01..0000000
--- a/packages/block-editor/src/components/page-template-picker/container.native.js
+++ /dev/null
@@ -1,6 +0,0 @@
-/**
- * External dependencies
- */
-import { View } from 'react-native';
-
-export default View;
diff --git a/packages/block-editor/src/components/page-template-picker/picker.js b/packages/block-editor/src/components/page-template-picker/picker.js
index 886c649..68d71d3 100644
--- a/packages/block-editor/src/components/page-template-picker/picker.js
+++ b/packages/block-editor/src/components/page-template-picker/picker.js
@@ -9,21 +9,18 @@ import { withDispatch } from '@wordpress/data';
  */
 import __experimentalBlockListFooter from '../block-list-footer';
 import Button from './button';
-import Container from './container';
 import defaultTemplates from './default-templates';
 
 const __experimentalPageTemplatePicker = ( { templates = defaultTemplates, resetContent } ) => {
 	return (
 		<__experimentalBlockListFooter>
-			<Container style={ { flexDirection: 'row' } }>
-				{ templates.map( ( { name, content } ) => (
-					<Button
-						key={ name }
-						onPress={ () => resetContent( content ) }
-						label={ name }
-					/>
-				) ) }
-			</Container>
+			{ templates.map( ( { name, content } ) => (
+				<Button
+					key={ name }
+					onPress={ () => resetContent( content ) }
+					label={ name }
+				/>
+			) ) }
 		</__experimentalBlockListFooter>
 	);
 };
```
</details>

But I wasn't sure if there were other _future_ reasons for the container here. Wdyt?

Reply:
I think the horizontal layout is specific to the picker, which kind of looks like a toolbar. The block footer is an agnostic container and I would expect that if I added several components there they would stack vertically by default.

I don't think there's a platform divergence here in terms of UX, it just happens that on the web we get the layout we want for free because Button seems to default to `display: inline-flex` and they stack horizontally. We don't have a clear vision of what the UX would be for this on the web, but for now the container approach seems to me like the more semantically correct one.

Reply:
By "platform divergence", I just meant the two `container{.native,}.js` files, and that for web it's just a fragment (props silently ignored), and mobile it's a `View`, and can accept all View props.

The possibility of additional components in the BlockListFooter, and the expectation they'd have `flex-direction: column;` makes sense to me. Thanks for the explanation! I suppose I'd consider that a future reason :) (and I agree the UX is not clear yet). :+1: